java-codegen/src/main/resources/org/opensearch/client/codegen/templates/ObjectShape/Equals.mustache

```diff
@@ -0,0 +1,27 @@
+    public boolean equals(Object o) {
+        {{#extendsOtherShape}}
+            if(!super.equals(o)) {
+                return false;
+            }
+        {{/extendsOtherShape}}
+        if(this == o) return true;
+        if(this.getClass() != o.getClass()) return false;
+        {{^hasFieldsToSerialize}}
+        return true;
+        {{/hasFieldsToSerialize}}
+        {{#hasFieldsToSerialize}}
+            {{className}} other = ({{className}})o;
+            return
+            {{#fields}}
+                {{#isUnboxedPrimitive}}
+                    this.{{name}}() == other.{{name}}()
+                {{/isUnboxedPrimitive}}
+                {{^isUnboxedPrimitive}}
+                    java.util.Objects.equals(this.{{name}}, other.{{name}})
```

Comment:
The way to reference the defined type in `Types` and get it added to the imports is like so:
```suggestion
                    {{TYPES.Java.Util.Objects}}.equals(this.{{name}}, other.{{name}})
```